src/typespec.ts

```diff
@@ -38,6 +39,7 @@ export function makeTypeSpecFromSwaggerType(
     properties: undefined,
     hasAdditionalProperties: false,
     additionalPropertiesType: undefined,
```

Comment:
Curious if you know what ends up in these additional properties fields

Reply:
Dug into this one a bit but feeling particularly enlightened yet. This property comes from the [JSON schema specification](https://json-schema.org/draft/2019-09/json-schema-core.html#additionalProperties) and it looks like implementation typically results in creating a union type with the current type where the property names of the former aren't explicitly specified. Mostly derived that behavior from looking through the tests:

i.e. 
the definition
```
"test_add_props_04": { "type": "object", "properties": { "some_prop": { "type": "string" } }, "additionalProperties": { "type": "string" } }
```

becomes
```
export type test_add_props_04 = {
    'some_prop' ? : string;
} & {
    [key: string]: string;
};
```

Reply:
Got it, so maybe it is just for specifying a `Record<string, T>` type or something.

Reply:
Sometimes. You can also do something like:
```
```